exercises/grep/example.go

```diff
@@ -0,0 +1,102 @@
+package grep
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type grepOptions struct {
+	n_line_numbers      bool
+	l_file_only         bool
+	i_any_case          bool
+	v_invert            bool
+	x_match_entire_line bool
+	prefix_filenames    bool
+}
+
+// GrepFiles searches for pattern in the files given, using optional flags.
+func GrepFiles(pattern string, flags []string, files []string) []string {
+	output := []string{}
+	options := getOptions(flags, len(files))
+	for _, file := range files {
+		output = append(output, grepOneFile(pattern, file, options)...)
+	}
+	return output
+}
+
+// grepOneFile performs a search for pattern in the given file
+// and returns the proper output according to the options.
```

Comment:
`s/proper/matched/` ?

Reply:
s/proper output/output/
Since the output depends on the options and not necessarily matched lines.